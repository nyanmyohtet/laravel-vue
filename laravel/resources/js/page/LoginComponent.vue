<template>
  <div>
    <h1>Larave Vue</h1>
    <form>
      <label for="email">
        email:
        <input type="text" name="email" v-model="email" />
      </label>
      <label>
        password:
        <input type="password" name="password" v-model="password" />
      </label>
      <input
        type="submit"
        name="login"
        value="Login"
        v-on:click.prevent="loginHandler"
      />
    </form>
    <router-link to="/register">Register</router-link>
  </div>
</template>

<script src="../service/LoginService.js"></script>
