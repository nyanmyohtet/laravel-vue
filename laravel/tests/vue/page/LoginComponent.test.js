import { shallowMount } from "@vue/test-utils";
import LoginComponent from "../../../resources/js/page/LoginComponent.vue";

test("displays Login Component", () => {
  const wrapper = shallowMount(LoginComponent);

  // Assert the rendered text of the component
  expect(wrapper.text()).toContain("Login");
});

test("should match shallow mounted snapshot", () => {
  const wrapper = shallowMount(LoginComponent);

  // Assert the snapshot of the component
  expect(wrapper).toMatchSnapshot();
});
