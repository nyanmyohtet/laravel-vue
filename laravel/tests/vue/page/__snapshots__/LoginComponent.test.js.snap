// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match shallow mounted snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
